packages/angular/cli/src/commands/mcp/tools/examples.ts

```diff
@@ -0,0 +1,173 @@
+ * @param exampleDatabasePath The path to the SQLite database file containing the examples.
+ */
+export function registerFindExampleTool(server: McpServer, exampleDatabasePath: string): void {
+  let db: import('node:sqlite').DatabaseSync | undefined;
+  let queryStatement: import('node:sqlite').StatementSync | undefined;
+
+  server.registerTool(
+    'find_examples',
+    {
+      title: 'Find Angular Code Examples',
+      description:
+        'Before writing or modifying any Angular code including templates, ' +
+        '**ALWAYS** use this tool to find current best-practice examples. ' +
+        'This is critical for ensuring code quality and adherence to modern Angular standards. ' +
+        'This tool searches a curated database of approved Angular code examples and returns the most relevant results for your query. ' +
+        'Example Use Cases: ' +
+        "1) Creating new components, directives, or services (e.g., query: 'standalone component' or 'signal input'). " +
+        "2) Implementing core features (e.g., query: 'lazy load route', 'httpinterceptor', or 'route guard'). " +
+        "3) Refactoring existing code to use modern patterns (e.g., query: 'ngfor trackby' or 'form validation').",
+      inputSchema: {
+        query: z.string().describe(
+          `Performs a full-text search using FTS5 syntax. The query should target relevant Angular concepts.
+
+Key Syntax Features (see https://www.sqlite.org/fts5.html for full documentation):
+  - AND (default): Space-separated terms are combined with AND.
+    - Example: 'standalone component' (finds results with both "standalone" and "component")
+  - OR: Use the OR operator to find results with either term.
+    - Example: 'validation OR validator'
+  - NOT: Use the NOT operator to exclude terms.
+    - Example: 'forms NOT reactive'
+  - Grouping: Use parentheses () to group expressions.
+    - Example: '(validation OR validator) AND forms'
+  - Phrase Search: Use double quotes "" for exact phrases.
+    - Example: '"template-driven forms"'
+  - Prefix Search: Use an asterisk * for prefix matching.
+    - Example: 'rout*' (matches "route", "router", "routing")
+
+Examples of queries:
+  - Find standalone components: 'standalone component'
+  - Find ngFor with trackBy: 'ngFor trackBy'
+  - Find signal inputs: 'signal input'
+  - Find lazy loading a route: 'lazy load route'
+  - Find forms with validation: 'form AND (validation OR validator)'`,
+        ),
+      },
+      annotations: {
+        readOnlyHint: true,
+        openWorldHint: false,
+      },
+    },
+    async ({ query }) => {
+      if (!db || !queryStatement) {
+        suppressSqliteWarning();
+
+        const { DatabaseSync } = await import('node:sqlite');
+        db = new DatabaseSync(exampleDatabasePath, { readOnly: true });
+        queryStatement = db.prepare('SELECT * from examples WHERE examples MATCH ? ORDER BY rank;');
+      }
+
+      const sanitizedQuery = sanitizeSearchQuery(query);
```

Comment:
Is there no library we can use for sanitization, or do we just avoid another dependency? I'd prefer us not having to maintain this.

Reply:
Is there no library for this? It feels like logic that can be hard to revisit in the future

Reply:
no. It's not SQL. it's a custom microsyntax for the FTS5 extension that's being used as a string literal in the parent SQL query.